src/datarobotx/idp/batch_predictions.py

```diff
@@ -0,0 +1,173 @@
+                    - `table` : string,  the name of specified database table.
+                    - `schema` : string (optional), the name of specified database schema.
+                    - `catalog` : string (optional), (new in v2.22) the name of specified database
+                      catalog.
+                    - `statement_type` : string, the type of insertion statement to create,
+                      one of ``datarobot.enums.AVAILABLE_STATEMENT_TYPES``.
+                    - `update_columns` : list(string) (optional),  a list of strings containing
+                      those column names to be updated in case `statement_type` is set to a
+                      value related to update or upsert.
+                    - `where_columns` : list(string) (optional), a list of strings containing
+                      those column names to be selected in case `statement_type` is set to a
+                      value related to insert or update.
+                    - `credential_id` : string, the ID of the credentials holding information about
+                      a user with write-access to the JDBC data source (see
+                      :ref:`Credentials <credentials_api_doc>`).
+                    - `create_table_if_not_exists` : bool (optional), If no existing table is detected,
+                      attempt to create it before writing data with the strategy defined in the
+                      statementType parameter.
+    enabled: bool
+        Whether or not the definition should be active on a scheduled basis. If True, `schedule` is required
+    name: str
+        * Must be unique to your organization *
+        Name of batch prediction job definition. If given the name of an existing definition within the supplied
+        deployment (according to deployment_id), this function will overwrite that existing definition with parameters
+        specified in this function (batch_prediction_job, enabled, schedule).
+    schedule : dict (optional)
+        The ``schedule`` payload defines at what intervals the job should run, which can be
+        combined in various ways to construct complex scheduling terms if needed. In all of
+        the elements in the objects, you can supply either an asterisk ``["*"]`` denoting
+        "every" time denomination or an array of integers (e.g. ``[1, 2, 3]``) to define
+        a specific interval.
+
+    Returns
+    -------
+    id of the created/updated batch prediction job definition
+
+    """
+    dr.Client(token=token, endpoint=endpoint)  # type: ignore
+
+    jobs = BatchPredictionJobDefinition.list(search_name=name, deployment_id=deployment_id, limit=0)
+    job: BatchPredictionJobDefinition
+    if jobs:
+        # There should only be 1 batchpredictionjobdefinition with the same name.
+        job = jobs[0]
+        job.update(
+            enabled=enabled, batch_prediction_job=batch_prediction_job, name=name, schedule=schedule
```

Comment:
I can't remember if type hinting an argument as optional makes it default to None. If so schedule should be `schedule:Optional[Dict[str, Any]]=None`